### PR TITLE
Add drug stock fields to protocol drugs

### PR DESCRIPTION
**Pull request by vkrmis**
**Story card:** [ch2220](https://app.clubhouse.io/simpledotorg/story/2220/add-drug-category-and-stock-tracked-to-protocoldrug-data-model-and-ui)
[PRD](https://docs.google.com/document/d/13bPC3CjWWbaNV2D_hVD3Flu7A4NUWB5DdNVo-MX8i6Y/edit#heading=h.2tfhwc7ycgb)

## Because

We need to enrich the protocol drugs data model with some new fields, in preparation for Drug Stock work 

## This addresses

Adds `drug_category` and `stock_tracked` to `ProtocolDrug`s 

Screenshot 2021-01-11 at 7 29 58 PM

### CODE REVIEW

File: app/models/protocol_drug.rb [hunk 1]
```diff
@@ -5,6 +5,15 @@ class ProtocolDrug < ApplicationRecord
   validates :name, presence: true
   validates :dosage, presence: true
 
+  enum drug_categories: {
```
Comment (harimohanraj89) [inline app/models/protocol_drug.rb:8]: `drug_category` singular? Does plural work too?

File: app/views/admin/protocol_drugs/_form.html.erb [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
-      <%= form.text_field :name, id: :protocol_drug_name %>
-      <%= form.text_field :dosage, id: :protocol_drug_dosage %>
-      <%= form.text_field :rxnorm_code, id: :protocol_drug_rxnorm_code, label: "RxNorm code" %>
-
-      <%= form.primary %>
-    <% end %>
-</div>
\ No newline at end of file
+  <%= bootstrap_form_with(model: [:admin, @protocol, @protocol_drug], local: true, label_errors: true) do |form| %>
+    <%= form.text_field :name, id: :protocol_drug_name %>
```
Comment (harimohanraj89) [inline app/views/admin/protocol_drugs/_form.html.erb:3]: Shall we make the other design changes to this form while we're here? Namely:
* Name, dosage, and Rxnorm on a single line
* "Type" under that
* "Track" under that

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
Couple thoughts.

File: app/models/protocol_drug.rb (hunk 1 above)
Comment (vkrmis) [reply app/models/protocol_drug.rb:8]: Shouldn't, fixing.

Comment (vkrmis) [reply app/models/protocol_drug.rb:8]: Plural accidentally worked for what I was using it for 🤦 
`ProtocolDrug.drug_categories`

File: app/controllers/admin/protocol_drugs_controller.rb [hunk 3]
```diff
@@ -56,7 +56,9 @@ def protocol_drug_params
     params.require(:protocol_drug).permit(
       :name,
       :dosage,
-      :rxnorm_code
+      :rxnorm_code,
+      :drug_category,
+      :stock_tracked
```
Comment (ssrihari) [inline app/controllers/admin/protocol_drugs_controller.rb:61]: From the PRD, it seems like we want this to restrict what is seen on the dashboard. Does it also restrict the drugs that can be tracked? 

If we don't want to restrict the data-entry, maybe we should rename the field to reflect viewing, or reporting, something on those lines. `stock_tracked_on_dashboard`?

Comment (ssrihari) [reply app/controllers/admin/protocol_drugs_controller.rb:61]: Vikram explained over a conversation. Seems like we want to keep the two concerns of viewing-on-dashboard, and collecting-data as the same thing for now.

We can revisit splitting these two concerns later, but that would involve a data migration. 

cc @harimohanraj89 @kitallis

Comment (harimohanraj89) [reply app/controllers/admin/protocol_drugs_controller.rb:61]: Sounds good 👍

**Review by harimohanraj89 — APPROVED** [review]
Coooolbeans.

**Pull request merged**